core/src/main/cpp/gdn/gast_node.cpp

```diff
@@ -491,39 +297,16 @@ bool GastNode::intersects_ray(Vector3 ray_origin, Vector3 ray_direction, Vector3
     return plane.intersects_ray(ray_origin, ray_direction, intersection);
 }
 
-String GastNode::generate_shader_code() const {
-    String shader_code = kShaderCode;
-    if (render_on_top) {
-        shader_code += kDisableDepthTestRenderMode;
-    }
-    if (projection_mesh_type == ProjectionMeshType::EQUIRECTANGULAR) {
-        shader_code += kCullFrontRenderMode;
-    }
-    return shader_code;
-}
-
 Vector2 GastNode::get_relative_collision_point(Vector3 absolute_collision_point) {
-    Vector2 relative_collision_point = kInvalidCoordinate;
-
-    // Turn the collision point into local space
     Vector3 local_point = to_local(absolute_collision_point);
-
-    // Normalize the collision point. A Gast node is a flat quad so we only worry about
-    // the x,y coordinates
-    Vector2 node_size = get_size();
-    if (node_size.width > 0 && node_size.height > 0) {
-        float max_x = node_size.width / 2;
-        float min_x = -max_x;
-        float max_y = node_size.height / 2;
-        float min_y = -max_y;
-        relative_collision_point = Vector2((local_point.x - min_x) / node_size.width,
-                                           (local_point.y - min_y) / node_size.height);
-
-        // Adjust the y coordinate to match the Android view coordinates system.
-        relative_collision_point.y = 1 - relative_collision_point.y;
+    if (get_projection_mesh()->is_rectangular_projection_mesh()) {
+        auto *rectangular_projection_mesh =
+            dynamic_cast<RectangularProjectionMesh*>(get_projection_mesh());
```

Comment:
The `*ProjectionMesh` classes being derived from Godot's resources means you can use Godot's `Object::cast_to<RectangularProjectionMesh>(...)` method.

Reply:
Done.